### PR TITLE
Make export configurable

### PR DESCRIPTION
**Pull request by jenshnielsen**
This makes it possible to configure the exported name. So far it is possible to include captured_run_id and guid. 

- [x] Should we include other elements now or wait
- [x] Should we change the default to include guid?

This is technically an api break since it now uses captured_run_id and not run_id but I do think that is for the better.

### CODE REVIEW

**Comment by astafan8** [comment]
>  Should we include other elements now or wait

i'd vote for including :) at least smaple name, exp name, ds name, started ts .

> Should we change the default to include guid?

definitely!

> This is technically an api break since it now uses captured_run_id and not run_id but I do think that is for the better.

agree! we could consider this a bug fix.

File: qcodes/configuration/qcodesrc_schema.json [hunk 1]
```diff
@@ -342,6 +342,15 @@
                     "type": "string",
                     "default": "qcodes_",
                     "description": "Prefix to add to filename for exporting datasets to disk after a measurement finishes. Only activated if export_type is set"
+                },
+                "export_name_elements": {
+                    "description": "elements to use in export name. Names will be formated like '{export_prefix}_{export_name_elements[0]}_{...}.file_extension}",
```
Comment (astafan8) [inline qcodes/configuration/qcodesrc_schema.json:347]: what about allowing folks to provide a format string that use the "elements"? the default would be ``{prefix}_{captured_run_id}_{guid}`` (or so) and the format string give full flexibility for users and reliefs us from maintaining the conversion from "elements" to "post_fix" :) what do you think @jenshnielsen ?

Comment (jenshnielsen) [reply qcodes/configuration/qcodesrc_schema.json:347]: I am not really sure. It seems like one more thing that users have to get right and if they don't exporting will throw an exception. I would thing that being able to control the order of the elements should be enough

Comment (astafan8) [reply qcodes/configuration/qcodesrc_schema.json:347]: ok, thats fair. then let's go with the elements for now :)

**Comment by jenshnielsen** [comment]
bors merge

**Comment by jenshnielsen** [comment]
need to write a changelog entry

**Comment by jenshnielsen** [comment]
bors merge

**Comment by jenshnielsen** [comment]
bors merge

**Comment by jenshnielsen** [comment]
bors merge

**Pull request merged**